homeassistant/components/xiaomi_miio/sensor.py

```diff
@@ -142,19 +135,29 @@ def device_state_attributes(self):
             _LOGGER.debug("Got new state: %s", state)
 
             self._available = True
-            self._state = state.aqi
-            self._state_attrs.update(
-                {
-                    ATTR_POWER: state.power,
-                    ATTR_CHARGING: state.usb_power,
-                    ATTR_BATTERY_LEVEL: state.battery,
-                    ATTR_DISPLAY_CLOCK: state.display_clock,
-                    ATTR_NIGHT_MODE: state.night_mode,
-                    ATTR_NIGHT_TIME_BEGIN: state.night_time_begin,
-                    ATTR_NIGHT_TIME_END: state.night_time_end,
-                    ATTR_SENSOR_STATE: state.sensor_state,
-                }
-            )
+            if self._model != MODEL_AIRQUALITYMONITOR_B1:
+                self._state = state.aqi
+                self._unit_of_measurement = 'AQI'
+
+                self._state_attrs.update(
+                    {
+                        ATTR_POWER: state.power,
+                        ATTR_CHARGING: state.usb_power,
+                        ATTR_BATTERY_LEVEL: state.battery,
+                        ATTR_DISPLAY_CLOCK: state.display_clock,
+                        ATTR_NIGHT_MODE: state.night_mode,
+                        ATTR_NIGHT_TIME_BEGIN: state.night_time_begin,
+                        ATTR_NIGHT_TIME_END: state.night_time_end,
+                        ATTR_SENSOR_STATE: state.sensor_state,
+                    }
+                )
+            else:
+                """ As AQI is not available, temperature will be the state value
+                    TVOC is always in µg/m³, no matter the settings in the device
+                """
+                self._state = state.temperature
+                self._unit_of_measurement = '°C'
+                self._state_attrs.update(state.data)
```

Comment:
What are the state attributes? We prefer to set these explicitly to allow us to see what they are and how they are formatted.